### PR TITLE
ddl: args v2 for lock/unlock/repair table and recover table/schema

### PR DESCRIPTION
**Pull request by joechenrh**
### What problem does this PR solve?

Issue Number: ref #53930

Problem Summary:

### What changed and how does it work?

As title.

To resolve cycle import, `meta.AutoIDGroup`, `ddl.RecoverInfo` and `ddl.RecoverSchemaInfo` have been moved into `model` package.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/meta/model/job_args_test.go [hunk 1]
```diff
@@ -613,3 +613,69 @@ func TestCheckConstraintArgs(t *testing.T) {
 		require.True(t, args.Enforced)
 	}
 }
+
+func TestLockTableArgs(t *testing.T) {
+	inArgs := &LockTablesArgs{
+		LockTables:    []TableLockTpInfo{{1, 1, model.TableLockNone}},
+		UnlockTables:  []TableLockTpInfo{{2, 2, model.TableLockNone}},
+		IndexOfLock:   0,
+		IndexOfUnlock: 0,
```
Comment (D3Hunter) [inline pkg/meta/model/job_args_test.go:622]:
```suggestion
		IndexOfLock:   34,
		IndexOfUnlock: 123,
```
zero is hard to difference with default

File: pkg/ddl/schema_version.go [hunk 2]
```diff
@@ -266,17 +266,12 @@ func SetSchemaDiffForCreateTable(diff *model.SchemaDiff, job *model.Job) error {
...
-		recoverSchemaInfo      *RecoverSchemaInfo
-		recoverSchemaCheckFlag int64
-	)
-	err := job.DecodeArgs(&recoverSchemaInfo, &recoverSchemaCheckFlag)
+	args, err := model.GetRecoverArgs(job)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Reserved recoverSchemaCheckFlag value for gc work judgment.
-	job.Args[checkFlagIndexInJobArgs] = recoverSchemaCheckFlag
```
Comment (joccau) [inline pkg/ddl/schema_version.go:278]: Reserved recoverSchemaCheckFlag ?

**Review by joccau — COMMENTED** [review]
rest lgtm

File: pkg/ddl/schema_version.go (hunk 2 above)
Comment (joechenrh) [reply pkg/ddl/schema_version.go:278]: I don't know why it decode `recoverSchemaCheckFlag` and set it again here. I'll remove the comment here.

Comment (D3Hunter) [reply pkg/ddl/schema_version.go:278]: might because it became a pointer after DecodeArgs

**Review by D3Hunter — APPROVED** [review]

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**